### PR TITLE
Gecodeerde lithologie

### PR DESCRIPTION
**Pull request by pjhaest**
* [x] I have read and followed the guidelines in the `CONTRIBUTING` document
* [x] I have checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.

Gecodeerde lithologie search method works fine in Python 2.7. Documentation will follow.

### CODE REVIEW

File: pydov/types/interpretaties.py [hunk 1]
```diff
@@ -344,3 +344,92 @@ class LithologischeBeschrijvingen(AbstractBoringInterpretatie):
...
+
+    _name = 'gecodeerde_lithologie_laag'
+    _rootpath = './/gecodeerdelithologie/laag'
+
+    _fields = [{
+        'name': 'diepte_laag_van',
+        'source': 'xml',
+        'sourcefield': '/van',
+        'definition': 'Diepte van de bovenkant van de laag lithologische'
+                      ' beschrijving in meter.',
```
Comment (Roel) [inline pydov/types/interpretaties.py]: This needs to be "Diepte van de bovenkant van de laag gecodeerde lithologie in meter"

File: pydov/types/interpretaties.py [hunk 2]
```diff
@@ -344,3 +344,92 @@ class LithologischeBeschrijvingen(AbstractBoringInterpretatie):
...
+        'type': 'float',
+        'notnull': False
+    }, {
+        'name': 'diepte_laag_tot',
+        'source': 'xml',
+        'sourcefield': '/tot',
+        'definition': 'Diepte van de onderkant van de laag lithologische'
+                      ' beschrijving in meter.',
```
Comment (Roel) [inline pydov/types/interpretaties.py]: This needs to be "Diepte van de onderkant van de laag gecodeerde lithologie in meter"

File: pydov/types/interpretaties.py [hunk 3]
```diff
@@ -344,3 +344,92 @@ class LithologischeBeschrijvingen(AbstractBoringInterpretatie):
...
+        'definition': 'grondsoort (als code) van de laag gecodeerde '
+                      'lithologie',
+        'type': 'string',
+        'notnull': False
+    }, {
+        'name': 'bijmenging_plaatselijk',
+        'source': 'xml',
+        'sourcefield': '/bijmenging/plaatselijk',
+        'definition': 'plaatselijk of niet-plaatselijk',
+        'type': 'string',
```
Comment (Roel) [inline pydov/types/interpretaties.py]: This needs to be 'boolean'

File: pydov/types/interpretaties.py [hunk 4]
```diff
@@ -344,3 +344,92 @@ class LithologischeBeschrijvingen(AbstractBoringInterpretatie):
...
+        'definition': 'Diepte van de onderkant van de laag lithologische'
+                      ' beschrijving in meter.',
+        'type': 'float',
+        'notnull': False
+    }, {
+        'name': 'hoofd_grondsoort',
+        'source': 'xml',
+        'sourcefield': '/hoofdnaam/grondsoort',
+        'definition': 'grondsoort (als code) van de laag gecodeerde '
+                      'lithologie',
```
Comment (Roel) [inline pydov/types/interpretaties.py]: Maybe add something from the XSD definition of 'hoofdnaam': "Lithologisch hoofdbestandeel van de laag gecodeerde lithologie"

File: pydov/types/interpretaties.py [hunk 5]
```diff
@@ -344,3 +344,92 @@ class LithologischeBeschrijvingen(AbstractBoringInterpretatie):
...
+        'definition': 'aanduiding van de hoeveelheid bijmenging',
+        'type': 'string',
+        'notnull': False
+    }, {
+        'name': 'bijmenging_grondsoort',
+        'source': 'xml',
+        'sourcefield': '/bijmenging/grondsoort',
+        'definition': 'type grondsoort (als code) van de laag '
+                      'gecodeerde lithologie of geotechnische '
+                      'codering',
```
Comment (Roel) [inline pydov/types/interpretaties.py]: Maybe add something from the XSD definition of 'bijmenging': "lithologisch nevenbestanddeel (als code) van de laag gecodeerde lithologie"

**Comment by Roel** [comment]
Additionally, we need to think about whether one `hoofdnaam` and one `bijmenging` are sufficient. There can be up to two `hoofdnaam` and up to three `bijmenging` elements. If we want to include all of them, we need to choose how we do this in the output dataframe: either as additional columns or as additional rows (flattening the model as we do now already for interpretatie -> laag).

'Gecodeerde lithologie' interpretations can have extra details besides the `hoofdnaam` and `bijmenging` elements: f.ex. `fossiel`, `secundairkenmerk`, `sedimentairestructuur`, `ader`, `tektoniek`. Are any of these relevant for inclusion or did we discuss this already when defining the default output dataframe?

Also, I had rather discussed this before it went into master.. I cannot check PRs within two hours after they are opened. Please don't merge your own PRs without (leaving some time for) discussion. Thanks!

**Comment by pjhaest** [comment]
@Roel : ok, will leave PR's open for review in the future, was happily surprised by the all green on the checks ;-)

Regarding `hoofdnaam` and `bijmenging`: I would keep it to flatten with additional rows as for 'interpretatie --> laag'

Regarding the extra details: I have no need of these in the search method 'gecodeerde lithologie'. But maybe others can benefit (altough I think this information is already included in 'lithologische beschrijvingen')

**Pull request merged**